### ISSUE TITLE
Disable node selection "direct" and "delayed" while the map is scrolled

### ISSUE THREAD

**Issue by aaa1386** [freeplane/freeplane#1925]
If the node selection mode is set in the two modes `Preferences -> Behavior -> Selection method -> Delayed or Direct` , a problem occurs:
When we scroll the map with the mouse or when the map is scrolled automatically, if the mouse pointer is placed on a node, that node is selected.
The user only intends to scroll the map, but the nodes are selected unintentionally.

If possible make a change in FP so that nodes are not randomly selected when the map is scrolling.

**Comment by dpolivaev** [comment]
Preview 1.12.2_10

**Comment by aaa1386** [comment]
> Preview 1.12.2_10

Thank you.
There is still a bug.
When scrolling stops, whatever node the mouse cursor is on is selected. This choice is not the user's choice, but it is randomly overlapped with the mouse node after scrolling that node and is automatically selected.

This bug only exists in manual scroll mode , not automatic scroll mode and auto-expand.
It means that now there are two different policies in FP.

**Comment by aaa1386** [comment]
In addition, there is a mistake in choosing the node. It is better to remove this automatic selection.


https://github.com/user-attachments/assets/1fd532b1-cc48-4916-969a-ae43412f7e2d

**Comment by euu2021** [comment]
I use the Direct selection method and I noticed that the selection got slower, in some specific situations, when I moved from freeplane-1.12.2-pre09 to freeplane-1.12.2-alpha-pre11. The problem seems to be related to changing the zoom level.

So, when I change the zoom level, the selection gets unresponsive for a moment:

![javaw_uMMRizoyxG](https://github.com/user-attachments/assets/a285f810-1f4d-4409-ab6e-97344217ff33)

**Comment by dpolivaev** [comment]
@euu2021 The change is intentional: if the map is scrolled by any reason, the mouse pointer can point at accidental node. Therefore direct and delayed selection is disabled for about half a second after the scrolling.

Use case: you want to scroll map by iterative dragging and do not want to select and nodes when you move the mouse to drag it.

**Comment by dpolivaev** [comment]
Preview 1.12.2_12 alpha

**Comment by aaa1386** [comment]
> Preview 1.12.2_12 alpha

OK

**Comment by euu2021** [comment]
> @euu2021 The change is intentional: if the map is scrolled by any reason, the mouse pointer can point at accidental node. Therefore direct and delayed selection is disabled for about half a second after the scrolling.
> 
> Use case: you want to scroll map by iterative dragging and do not want to select and nodes when you move the mouse to drag it.

I prefer the old behavior. The half a second freeze after the scrolling is annoying. It slows me down.

**Comment by dpolivaev** [comment]
I tried to find a compromise, preview 1.12.2_13

**Comment by aaa1386** [comment]
> I tried to find a compromise, preview 1.12.2_13


I did not understand the compromise.
Look, I will explain what upset @euu2021.
 At the moment when the mouse pointer is placed on a node after scrolling, why is the node on which the mouse pointer is placed not selected?
When the mouse is moving it's fine if the nodes are not selected in a row and it won't cause any problems for anyone. Because it doesn't make sense to select nodes between paths as long as it is scrolling.

Therefore, the dispute is where the mouse pointer is placed on a node after scrolling. Now the mouse needs to make a small movement to wake up to select the node. I do not insist on this small movement. My purpose of the video was more to show an error in node selection. It showed that the mouse pointer is placed in the node, that node is not selected, but the higher node is selected.
I also came to the conclusion with the reviews that after the scroll is finished, it is better if we don't need to move the mouse to wake up.
There is a reason for this: when the user is scrolling, he can also think about where to stop scrolling. It can stop the mouse at the node that wants to be selected. So the place that stops scrolling can be the node that wants to be selected. So it should not try to select the node again. Although it is usually difficult to scroll and be accurate.

 Of course, I would like the amount of pause that the user has set for the "Delayed" mode to select the mouse to be applied.
It is clear that the "direct" mode does not require delay.
